### ISSUE TITLE
Attempting to start 5.201 reaches time out with 'could not join cluster' error

### ISSUE THREAD

**Issue by Dougnlizt** [payara/Payara#4670]
# Description #
----------

Migrating from 5.191 to 5.201.  5.191 domain starts fine.  5.201 always times out.  Linux only - Windows 10 working fine.

## Expected Outcome

When running asadmin start-domain, I expected the payara server to start.

## Current Outcome

After the 600 timeout is reached, it says the server failed to start.  First message noted says "Could not join cluster."

## Steps to reproduce (Only for bug reports) 

I tried to find any other report similar to my issue.  So far cannot find anything at all.  It seems like it is probably something simple.

1 -**  Download and unzip Payara 5.201

2 -**  Start the domain

    ./asadmin start-domain --debug --verbose

3 -**  After 5 minutes, it bails with error info

    Here is the error output from the log file:  

>   JVM invocation command line:
/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.191.b12-1.el7_6.x86_64/jre/bin/java
-cp
/home/linuxuser/Payara/payara-5.201/glassfish/modules/glassfish.jar
-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=9009
-XX:+UnlockDiagnosticVMOptions
-XX:NewRatio=2
-Xbootclasspath/p:/home/linuxuser/Payara/payara-5.201/glassfish/lib/grizzly-npn-bootstrap-1.8.1.jar
-Xmx512m
-server
-javaagent:/home/linuxuser/Payara/payara-5.201/glassfish/lib/monitor/flashlight-agent.jar
-Djavax.xml.accessExternalSchema=all
-Djavax.net.ssl.trustStore=/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1/config/cacerts.jks
-Djdk.tls.rejectClientInitiatedRenegotiation=true
-Djdk.corba.allowOutputStreamSubclass=true
-Dfelix.fileinstall.dir=/home/linuxuser/Payara/payara-5.201/glassfish/modules/autostart/
-Dorg.glassfish.additionalOSGiBundlesToStart=org.apache.felix.shell,org.apache.felix.gogo.runtime,org.apache.felix.gogo.shell,org.apache.felix.gogo.command,org.apache.felix.shell.remote,org.apache.felix.fileinstall
-Dcom.sun.aas.installRoot=/home/linuxuser/Payara/payara-5.201/glassfish
-Dfelix.fileinstall.poll=5000
-Djava.security.policy=/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1/config/server.policy
-Djava.endorsed.dirs=/home/linuxuser/Payara/payara-5.201/glassfish/modules/endorsed:/home/linuxuser/Payara/payara-5.201/glassfish/lib/endorsed
-Dosgi.shell.telnet.maxconn=1
-Dfelix.fileinstall.bundles.startTransient=true
-Dcom.sun.enterprise.config.config_environment_factory_class=com.sun.enterprise.config.serverbeans.AppserverConfigEnvironmentFactory
-Dfelix.fileinstall.log.level=2
-Djavax.net.ssl.keyStore=/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1/config/keystore.jks
-Djava.security.auth.login.config=/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1/config/login.conf
-Dfelix.fileinstall.disableConfigSave=false
-Dorg.glassfish.grizzly.DEFAULT_MEMORY_MANAGER=org.glassfish.grizzly.memory.HeapMemoryManager
-Dfelix.fileinstall.bundles.new.start=true
-Dcom.sun.aas.instanceRoot=/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1
-Dosgi.shell.telnet.port=6666
-Dgosh.args=--nointeractive
-Dcom.sun.enterprise.security.httpsOutboundKeyAlias=s1as
-Dorg.jboss.weld.serialization.beanIdentifierIndexOptimization=false
-Dosgi.shell.telnet.ip=127.0.0.1
-DANTLR_USE_DIRECT_CLASS_LOADING=true
-Djava.awt.headless=true
-Dcom.ctc.wstx.returnNullForDefaultNamespace=true
-Djava.ext.dirs=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.191.b12-1.el7_6.x86_64/jre/lib/ext:/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.191.b12-1.el7_6.x86_64/jre/jre/lib/ext:/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1/lib/ext
-Djdbc.drivers=org.h2.Driver
-Dorg.glassfish.grizzly.nio.DefaultSelectorHandler.force-selector-spin-detection=true
-Djava.library.path=/home/linuxuser/Payara/payara-5.201/glassfish/lib:/usr/java/packages/lib/amd64:/usr/lib64:/lib64:/lib:/usr/lib
com.sun.enterprise.glassfish.bootstrap.ASMain
-upgrade
false
-domaindir
/home/linuxuser/Payara/payara-5.201/glassfish/domains/domain1
-read-stdin
true
-asadmin-args
--host,,,localhost,,,--port,,,4848,,,--secure=false,,,--terse=false,,,--extraterse=false,,,--echo=false,,,--interactive=true,,,--autoname=false,,,start-domain,,,--verbose=true,,,--watchdog=false,,,--debug=true,,,--domaindir,,,/home/linuxuser/Payara/payara-5.201/glassfish/domains,,,domain1
-domainname
domain1
-instancename
server
-type
DAS
-verbose
true
-asadmin-classpath
/home/linuxuser/Payara/payara-5.201/glassfish/lib/client/appserver-cli.jar
-debug
true
-asadmin-classname
com.sun.enterprise.admin.cli.AdminMain
-watchdog
false]]

> 
> [2020-05-15T11:37:56.163-0600] [Payara 5.201] [INFO] [NCLS-LOGGING-00009] [javax.enterprise.logging] [tid: _ThreadID=28 _ThreadName=RunLevelControllerThread-1589564276087] [timeMillis: 1589564276163] [levelValue: 800] [[
>   Running Payara Version: Payara Server  5.201 #badassfish (build 512)]]
> 
> [2020-05-15T11:37:56.167-0600] [Payara 5.201] [INFO] [NCLS-LOGGING-00010] [javax.enterprise.logging] [tid: _ThreadID=28 _ThreadName=RunLevelControllerThread-1589564276087] [timeMillis: 1589564276167] [levelValue: 800] [[
>   Server log file is using Formatter class: com.sun.enterprise.server.logging.ODLLogFormatter]]
> 
> [2020-05-15T11:37:56.531-0600] [Payara 5.201] [INFO] [] [fish.payara.nucleus.hazelcast.HazelcastCore] [tid: _ThreadID=30 _ThreadName=RunLevelControllerThread-1589564276091] [timeMillis: 1589564276531] [levelValue: 800] [[
>   Data grid encryption is enabled]]
> 
> [2020-05-15T11:37:56.649-0600] [Payara 5.201] [INFO] [] [org.glassfish.ha.store.spi.BackingStoreFactoryRegistry] [tid: _ThreadID=28 _ThreadName=RunLevelControllerThread-1589564276087] [timeMillis: 1589564276649] [levelValue: 800] [[
>   Registered fish.payara.ha.hazelcast.store.HazelcastBackingStoreFactoryProxy for persistence-type = hazelcast in BackingStoreFactoryRegistry]]
> 
> [2020-05-15T11:42:57.871-0600] [Payara 5.201] [SEVERE] [] [com.hazelcast.instance.Node] [tid: _ThreadID=27 _ThreadName=RunLevelControllerThread-1589564276086] [timeMillis: 1589564577871] [levelValue: 1000] [[
>   [137.201.72.180]:4900 [development] [3.12.6] Could not join cluster. Shutting down now!]]
> 
> [2020-05-15T11:42:57.877-0600] [Payara 5.201] [WARNING] [] [com.hazelcast.instance.Node] [tid: _ThreadID=27 _ThreadName=RunLevelControllerThread-1589564276086] [timeMillis: 1589564577877] [levelValue: 900] [[
>   [137.201.72.180]:4900 [development] [3.12.6] Terminating forcefully...]]
> 
> [2020-05-15T11:42:57.902-0600] [Payara 5.201] [WARNING] [] [com.hazelcast.util.PhoneHome] [tid: _ThreadID=27 _ThreadName=RunLevelControllerThread-1589564276086] [timeMillis: 1589564577902] [levelValue: 900] [[
>   [137.201.72.180]:4900 [development] [3.12.6] Could not schedule phone home task! Most probably Hazelcast failed to start.]]
> 

> [2020-05-15T11:42:57.905-0600] [Payara 5.201] [INFO] [NCLS-CORE-00015] [javax.enterprise.system.core] [tid: _ThreadID=29 _ThreadName=RunLevelControllerThread-1589564276091] [timeMillis: 1589564577905] [levelValue: 800] [[
>   Shutdown requested
> MultiException stack 1 of 9
> java.lang.IllegalStateException: Node failed to start!
> 	at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:138)
> 	at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:229)
> ...
> 
> MultiException stack 2 of 9
> java.lang.IllegalStateException: Unable to perform operation: post construct on fish.payara.nucleus.config.ClusteredConfig
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:369)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
> 	at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:65)
> ...
> 	
> MultiException stack 3 of 9
> java.lang.IllegalArgumentException: While attempting to resolve the dependencies of fish.payara.nucleus.requesttracing.RequestTracingService errors were found
> 	at org.jvnet.hk2.internal.ClazzCreator.resolveAllDependencies(ClazzCreator.java:224)
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:334)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> ...
> 
> MultiException stack 4 of 9
> java.lang.IllegalStateException: Unable to perform operation: resolve on fish.payara.nucleus.requesttracing.RequestTracingService
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:363)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
> 	at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:65)
> ...
> 
> MultiException stack 5 of 9
> java.lang.IllegalArgumentException: While attempting to resolve the dependencies of fish.payara.monitoring.runtime.MonitoringConsoleRuntimeImpl errors were found
> 	at org.jvnet.hk2.internal.ClazzCreator.resolveAllDependencies(ClazzCreator.java:224)
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:334)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:59)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:47)
> 	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture$1.call(Cache.java:74)
> ...
> 
> MultiException stack 6 of 9
> java.lang.IllegalStateException: Unable to perform operation: resolve on fish.payara.monitoring.runtime.MonitoringConsoleRuntimeImpl
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:363)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:59)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:47)
> ...
> 
> MultiException stack 7 of 9
> java.lang.IllegalStateException: Unable to perform operation: post construct on com.sun.enterprise.v3.server.ApplicationLifecycle
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:363)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:59)
> 	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:47)
> 	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture$1.call(Cache.java:74)
> ...
> 
> MultiException stack 8 of 9
> java.lang.IllegalArgumentException: While attempting to resolve the dependencies of com.sun.enterprise.v3.server.ApplicationLoaderService errors were found
> 	at org.jvnet.hk2.internal.ClazzCreator.resolveAllDependencies(ClazzCreator.java:224)
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:334)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
> ...
> 
> MultiException stack 9 of 9
> java.lang.IllegalStateException: Unable to perform operation: resolve on com.sun.enterprise.v3.server.ApplicationLoaderService
> 	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:363)
> 	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:463)
> 	at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:281)
> 	at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:65)
> ...
> ]]
> 
> [2020-05-15T11:42:57.907-0600] [Payara 5.201] [SEVERE] [NCLS-CORE-00016] [javax.enterprise.system.core] [tid: _ThreadID=29 _ThreadName=RunLevelControllerThread-1589564276091] [timeMillis: 1589564577907] [levelValue: 1000] [[
>   Startup service failed to start

## Context

Payara 5.184, 5.191 both work fine.  5.201 works fine for me in my Windows 10 environment, so I was surprised with the problem I've been having in Linux.  I tried making modifications to the domain.xml file, but nothing seems to help.

## Environment ##

- **Payara Version**: 5.201
- **Edition**: Full
- **JDK Version**: OpenJDK 8 u191; Zulu OpenJDK 8 u252
- **Operating System**: CentOS Linux 7 (Kernel: Linux 3.10.0-957.5.1.el7.x86_64)

**Comment by janArb** [comment]
I can confirm this bug for ubuntu 14.04 with Zulu OpenJDK 8 u252

**Comment by janArb** [comment]
The very same download files starts fine in windows 10 with Payara 5.194 and Zulu OpenJDK 8 u252

**Comment by janArb** [comment]
I got Payara 5.191 running with Zulu OpenJDK 8 u252 on this ubuntu 14 machine now (as noted from the issue reporter above), I also tested 5.193.1 and that was also not working due to the same reason here.

**Comment by Dougnlizt** [comment]
Thanks for the update.  Similar to you, in our Linux environment, the following combination works for us:

- Payara 5.191
- Zulu OpenJDK 8 u252
- CentOS Linux 7

Once we go past Payara 5.191, we start having issues.

**Comment by smillidge** [comment]
Random thought try disabling VNC if you have that running as it could be a port clash with Hazelcast

**Comment by smillidge** [comment]
See https://github.com/payara/Payara/issues/3995

**Comment by janArb** [comment]
There was indeed running vnc4server at port 5900. However we tested with Payara 5.2020.2 and that is also starting without issues if the vnc server is running. Issue is solved.

**Comment by janArb** [comment]
Sorry this was a false positive, the server started only once. After that it didn't start until I changed the VNC server port, i.e. this issue is also open for Payara 5.2020.2.

**Comment by OndroMih** [comment]
I'll try to reproduce with 5.2020.4. It was reported here that the issue is still there with 5.2020.4: https://stackoverflow.com/questions/58521647/payara5-server-will-not-deploy-unknown-protocol-rfb

**Comment by MeroRai** [comment]
Hi @janArb,

Due to how long ago this issue was raised we have decided to close the issue immediately. If the issue is still pressing to you, feel free to verify if it’s applicable in the current release of Payara Community edition, and proceed to raise a new issue with details of the test reproducer. Many thanks for your understanding.

**Comment by janArb** [comment]
Hey @MeroRai , I won't test this again, because in the link of @OndroMih a workaround is documented that is working if shutting down VNC once:

Deactivate completely: 
asadmin set-hazelcast-configuration --enabled=false

Change port if it should stay active:
asadmin set-hazelcast-configuration --startport=5902

For me this is fine and the issue can stay closed.
